src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -279,6 +290,21 @@ public RequestBuilder(
     this.host = hostName;
     this.userAgentSuffix = userAgentSuffix;
 
+    // create our security/token manager
+    if (credential instanceof KeyPair) {
+      securityManager =
+          new JWTManager(accountName, userName, (KeyPair) credential, telemetryService);
+      authType = Constants.JWT;
+    } else if (credential instanceof OAuthCredential) {
+      securityManager =
+          new OAuthManager(
+              accountName, userName, (OAuthCredential) credential, makeBaseURI(), telemetryService);
+      authType = Constants.OAUTH;
```

Comment:
If we are assigning securityManager here could we simply call something like `securityManager.getAuthType()` and use when needed?

Reply:
It turns out that we actually don't need `authType`, removed.